### ISSUE TITLE
Support user-defined coordinate systems

### ISSUE THREAD

**Issue by dbaston** [mapserver/mapserver#5957]
MapServer currently returns an error if a map is requested in a non-EPSG coordinate system, e.g. `crs=user:54012`, even though such a coordinate system may be understood by PROJ:

> msWMSLoadGetMapParams(): WMS server error. Unsupported CRS namespace (only EPSG, AUTO2, CRS currently supported).

This was discussed previously (November 2018) on [mapserver-user](https://lists.osgeo.org/pipermail/mapserver-users/2018-November/080845.html). In that thread, it was proposed to allow support for a `USER` namespace.

If the consensus is that the CRS namespace really must be nothing other than "EPSG", "CRS", or "AUTO2" then an alternative approach, used by GeoServer and described [here](https://docs.geoserver.org/latest/en/user/configuration/crshandling/customcrs.html), would be to allow user additions into the EPSG namespace at the Mapfile level.

In the current situation, I think the only possible workaround would be to somehow define a new coordinate system by inserting it into the sqlite database PROJ uses for EPSG coordinate system definitions.

**Comment by swalterfub** [comment]
I strongly agree here, we need codes other than EPSG for planetary spatial reference systems. For example, see here for a GDAL-based solution: https://github.com/USGS-Astrogeology/GDAL_scripts/tree/master/OGC_IAU2000_WKT_v2
We need such solutions for providing planetary (other than Earth) map services! If not using WKT inside OGC services, we should have the possibility to use namespaces other than the ones mentioned.

**Comment by dbaston** [comment]
FWIW, the workaround I'm currently using is to copy PROJ strings for the EPSG codes I need into a a new file saved as `/usr/share/proj/epsg`, and then adding new codes for my own:

```
<2163> +proj=laea +lat_0=45 +lon_0=-100 +x_0=0 +y_0=0 +a=6370997 +b=6370997 +units=m +no_defs 
<3035> +proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
<3572> +proj=laea +lat_0=90 +lon_0=-150 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs 
<3857> +proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext  +no_defs
<4326> +proj=longlat +datum=WGS84 +no_defs 
<8857> +proj=eqearth +lon_0=0 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs
<54012> +proj=eck4 +lon_0=0 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs 

# Africa
<99901> +proj=laea +lon_0=18.6

# Australia, New Zealand
<99902> +proj=aea +lat_1=-36 +lat_2=-18 +lon_0=145
```

**Comment by swalterfub** [comment]
Yes, I'm doing the same. If you're using MapCache (which allows any arbitrary namespace), you can define additional "srsalias" codes for the one fixed srs. So the cache uses different namespace codes to the outside than internally to Mapserver.

**Comment by mirt001** [comment]
any updates on this one?
@swalterfub nice one, would that work for WCS? I am not familiar with MapCache, but [the documentation](https://mapserver.org/mapcache/) suggests that it only works for WMS
